### ISSUE TITLE
macOS: Opening the UI Editor tool crashes the Editor

### ISSUE THREAD

**Issue by LB-MateuszSzweda** [o3de/o3de#6569]
**Describe the bug**
Trying to open the UI Editor tool in the macOS version of the Editor results in a crash.

The issue was observed for AutomatedTesting and new projects.

For further information please refer to the attached video file and log files.

**Steps to reproduce**

1. Launch the Editor.
2. Create a new level.
3. Open the UI Editor tool via _Tools > UI Editor_
or by clicking the _UI Editor_ icon on the _Edit Mode Toolbar_

**Expected behavior**
The UI Editor tool opens and the Editor remains stable.

**Actual behavior**
The Editor crashes.

**Screenshots/Video**

https://user-images.githubusercontent.com/86953456/147477667-51b45176-1e11-455c-a8c3-cb8c49f70c0f.mov

**Found in Branch**
Development (c3de803)

**Desktop/Device:**
- Device: Mac
- OS: macOS
- Version: Monterey 12.0.1
- CPU: M1
- GPU: M1
- Memory: 16GB

**Additional context**
[crash.log](https://github.com/o3de/o3de/files/7780418/crash.log)
[Editor.log](https://github.com/o3de/o3de/files/7780419/Editor.log)

**Comment by amzn-sj** [comment]
The crash is happening in Atom RPI's pass system. From the logs:
```
[Error] (Pass System) - Pass::ProcessConnection - Pass [Root.MainPipeline_0.MainPipeline.ShadowPass.EsmShadowmapsPassDirectional] couldn't find a valid binding [Shadowmap] on pass [CascadedShadowmapsPass].
<14:37:18> [Error] (Pass System) - [Pass Exponentiation] Could not bind shader buffer index 'm_inputShadowmap' because it has no attachment.
<14:37:18> (PassSystem) - 
--- PASS VALIDATION FAILURE ---
--Critical Errors--
<14:37:18> (PassSystem) - 
There are 1 passes with errors:
<14:37:18> (PassSystem) - 
- EsmShadowmapsPassDirectional
   Pass::ProcessConnection - Pass [Root.MainPipeline_0.MainPipeline.ShadowPass.EsmShadowmapsPassDirectional] couldn't find a valid binding [Shadowmap] on pass [CascadedShadowmapsPass].
<14:37:18> (PassSystem) - 
There are 2 passes with missing Inputs:
<14:37:18> (PassSystem) - 
- EsmShadowmapsPassDirectional
   DepthShadowmaps has no valid attachment
<14:37:18> (PassSystem) - 
- Exponentiation
   Depth has no valid attachment
<14:37:18> (PassSystem) - 
There are 0 passes with missing Inputs/Outputs:
<14:37:18> (PassSystem) - 
--Non-Critical Errors/Warnings--
<14:37:18> (PassSystem) - 
There are 0 passes with warnings:
<14:37:18> (System) - 
==================================================================
<14:37:18> (System) - Trace::Assert
 /Users/srikkant/O3DE/o3de/Code/Framework/AzCore/AzCore/std/smart_ptr/intrusive_ptr.h(189): (8798813696) 'T *AZStd::intrusive_ptr<AZ::RPI::PassAttachment>::operator->() const [T = AZ::RPI::PassAttachment]'
<14:37:18> (System) - You can't dereference a null pointer
<14:37:18> (System) - ------------------------------------------------
<14:37:18> (System) - AZStd::intrusive_ptr<AZ::RPI::PassAttachment>::operator->() const (+0x63) [0x22356c8b3]
<14:37:18> (System) - AZ::Render::EsmShadowmapsPass::UpdateChildren() (+0x35) [0x2235e3d15]
<14:37:18> (System) - AZ::Render::EsmShadowmapsPass::FrameBeginInternal(AZ::RPI::Pass::FramePrepareParams) (+0x24) [0x2235e3c84]
<14:37:18> (System) - AZ::RPI::Pass::FrameBegin(AZ::RPI::Pass::FramePrepareParams) (+0x225) [0x24e1c53a5]
<14:37:18> (System) - AZ::RPI::ParentPass::FrameBeginInternal(AZ::RPI::Pass::FramePrepareParams) (+0x9f) [0x24e1c515f]
<14:37:18> (System) - AZ::RPI::Pass::FrameBegin(AZ::RPI::Pass::FramePrepareParams) (+0x225) [0x24e1c53a5]
<14:37:18> (System) - AZ::RPI::ParentPass::FrameBeginInternal(AZ::RPI::Pass::FramePrepareParams) (+0x9f) [0x24e1c515f]
<14:37:18> (System) - AZ::RPI::Pass::FrameBegin(AZ::RPI::Pass::FramePrepareParams) (+0x225) [0x7b9032725]
<14:37:18> (System) - AZ::RPI::ParentPass::FrameBeginInternal(AZ::RPI::Pass::FramePrepareParams) (+0x9f) [0x7b90324df]
<14:37:18> (System) - AZ::RPI::SwapChainPass::FrameBeginInternal(AZ::RPI::Pass::FramePrepareParams) (+0x148) [0x7b90a1fa8]
<14:37:18> (System) - AZ::RPI::Pass::FrameBegin(AZ::RPI::Pass::FramePrepareParams) (+0x225) [0x24e1c53a5]
<14:37:18> (System) - AZ::RPI::ParentPass::FrameBeginInternal(AZ::RPI::Pass::FramePrepareParams) (+0x9f) [0x24e1c515f]
<14:37:18> (System) - AZ::RPI::Pass::FrameBegin(AZ::RPI::Pass::FramePrepareParams) (+0x225) [0x24e1c53a5]
<14:37:18> (System) - AZ::RPI::PassSystem::FrameUpdate(AZ::RHI::FrameGraphBuilder&) (+0xcc) [0x24e1e86dc]
<14:37:18> (System) - AZ::RPI::RPISystem::RenderTick()::$_2::operator()(AZ::RHI::FrameGraphBuilder&) const (+0x2b) [0x24e65e02b]
<14:37:18> (System) - decltype(InvokeTraits::forward<AZ::RPI::RPISystem::RenderTick()::$_2&>(fp)(InvokeTraits::forward<AZ::RHI::FrameGraphBuilder&>(fp0))) AZStd::Internal::INVOKE<AZ::RPI::RPISystem::RenderTick()::$_2&, AZ::RHI::FrameGraphBuilder&>(AZ::RPI::RPISystem::RenderTic
<14:37:18> (System) - void AZStd::Internal::function_util::invoke_void_return_wrapper<void>::call<AZ::RPI::RPISystem::RenderTick()::$_2&, AZ::RHI::FrameGraphBuilder&>(AZ::RPI::RPISystem::RenderTick()::$_2&, AZ::RHI::FrameGraphBuilder&) (+0x32) [0x24e65df32]
<14:37:18> (System) - void AZStd::Internal::function_util::get_invoker<void (AZ::RHI::FrameGraphBuilder&), AZStd::Internal::function_util::function_obj_tag, AZStd::allocator>::call<AZ::RPI::RPISystem::RenderTick()::$_2>(AZStd::Internal::function_util::function_buffer&, AZ::RHI
<14:37:18> (System) - AZStd::function<void (AZ::RHI::FrameGraphBuilder&)>::operator()(AZ::RHI::FrameGraphBuilder&) const (+0x2d) [0x24e7874ad]
<14:37:18> (System) - AZ::RHI::RHISystem::FrameUpdate(AZStd::function<void (AZ::RHI::FrameGraphBuilder&)>) (+0x83) [0x24e787333]
<14:37:18> (System) - AZ::RPI::RPISystem::RenderTick() (+0x107) [0x24e60e1c7]
<14:37:18> (System) - AZ::RPI::RPISystemComponent::OnSystemTick() (+0x2a) [0x24df1055a]
<14:37:18> (System) - decltype(*(InvokeTraits::forward<AZ::Internal::HandlerNode<AZ::SystemTickEvents, AZ::SystemTickEvents, AZ::Internal::EBusContainer<AZ::SystemTickEvents, AZ::SystemTickEvents, (AZ::EBusAddressPolicy)0, (AZ::EBusHandlerPolicy)1>::HandlerHolder, false>&>(fp0
<14:37:18> (System) - ==================================================================
<14:37:18> (System) - ====Assert added to ignore list by spec and verbosity setting.====
```
I can work around it by adding a null check to prevent the crash in EsmShadowmapsPass::UpdateChildren(). But the EsmShadowmapsPassDirectional will still fail to find the binding between the attachments. This only seems to be a problem with EsmShadowmapsPassDirectional. May need someone from Atom to take a look.

**Comment by amzn-sj** [comment]
This PR is to prevent the crash: https://github.com/o3de/o3de/pull/6590
The underlying issue still needs to be addressed.

**Comment by amzn-sj** [comment]
The fix has been merged into development branch.

**Comment by LB-MateuszSzweda** [comment]
Verified as fixed in Development (5e8c1aa) . Opening the UI Editor tool no longer crashes the OSX Editor.
